File: solution_pool/14-longest-common-prefix/solution.cc
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
      string ret;
      if(strs.size()==0) {
        return ret;
      }
      if(strs.size()==1){
        return strs[0];
      }
      
      string base = strs[0];
      bool stop_flag = false;
      int k = -1;
      
      for(k=0; k<INT_MAX;++k) {
        for(int i=1; i< strs.size();++i) {  
          
          if(strs[i].size()<k) {
            return base.substr(0,k);
          }
          
          if(base[k] != strs[i][k]) {
            return base.substr(0,k);
      
          }
        }
      }
      
      
      
      
    }
};

File: solution_pool/14-longest-common-prefix/solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        ret=""
        if len(strs) == 0:
          return ret
        
        if len(strs) ==1:
          return strs[0]

        base = strs[0]     
        min_len=100000
        for m_str in strs:
          min_len = min(len(m_str),min_len)
          
        for i in range(min_len):
          for m_str in strs[1:]:
            if base[i]!=m_str[i]:
              return ret
          ret+=base[i]
        return ret
